Add support for field and header prefix

rgannu opened the pull request:
DBZ-2504: Added support for field and header prefix.

gunnarmorling commented on debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/transforms/ExtractNewDocumentState.java:
```diff
@@ -5,6 +5,19 @@
  */
 package io.debezium.connector.mongodb.transforms;
 
+import io.debezium.config.Configuration;
```
Seems like you applied some auto-organization of imports? Can you please revert this, to keep the diff focused on the actual changes? You can run `mvn process-sources` to ensure the formatting is right before committing. Thx!

rgannu replied:
yes.. My intellij reformat is doing that.. Fixed it now. Thanks

gunnarmorling replied:
No worries; looking much better now, thx!

gunnarmorling reviewed (changes requested):
Hey @rgannu, thx for this PR, but it's really hard to tell what the actual changes are with all the stylistic changes done at the same time. Please run the formatter (see inline comment), rework the commit and force-push. Thanks!

gunnarmorling reviewed (commented):
Thanks for the update, LGTM overall now, @rgannu! Could you only

* Add a test for the MongoDB implementation
* Add the new option to the docs under _documentation/modules/ROOT/pages/configuration/event-flattening.adoc_ and _documentation/modules/ROOT/pages/configuration/mongodb-event-flattening.adoc_, respectively

Once that's done, this is good to go. Thanks again!

rgannu commented:
> Thanks for the update, LGTM overall now, @rgannu! Could you only
> 
> * Add a test for the MongoDB implementation
> * Add the new option to the docs under _documentation/modules/ROOT/pages/configuration/event-flattening.adoc_ and _documentation/modules/ROOT/pages/configuration/mongodb-event-flattening.adoc_, respectively
> 
> Once that's done, this is good to go. Thanks again!

Updated the documentation and added test in the MongoDB implementation.

gunnarmorling commented:
Test failures are known and unrelated (tracked via https://issues.redhat.com/browse/DBZ-2268).

gunnarmorling commented:
Rebased and applied. Great work, thanks a lot, @rgannu!
